### PR TITLE
Parser fixes to get cmp(parse(s), parse(stringify(parse(s)))) === 0

### PR DESCRIPTION
**Pull request by mdittmer**
(0) Enable comparison tests
(1) Drop inconsistent custom compareTo()
(2) Fix up parse+stringify of 'inherit', serializer, type suffixes

### CODE REVIEW

**Comment by mdittmer** [comment]
Note: we need to land https://github.com/foam-framework/foam2/pull/368 and merge it into `foam2#beta-1` before this PR will go green.

File: lib/org/chromium/webidl/Parser.js [hunk 1]
```diff
@@ -525,7 +525,7 @@ foam.CLASS({
           if (v === ';') return parser.Serializer.create();
           return parser.Operation.isInstance(v) ?
               parser.Serializer.create({operation: v}) :
-          parser.Serializer.create({pattern: v});
+              parser.Serializer.create({pattern: v});
```
Comment (m-cheung) [inline lib/org/chromium/webidl/Parser.js:528]: nit: If we apply this formatting change here, we should apply it to the other ones around it for consistency.

**Review by arobins — COMMENTED** [review]
This all seems fine, waiting for it to go green before approving though.

**Review by m-cheung — COMMENTED** [review]
Aside from the minor nit, everything else looks good. I can verify that the test pass with changes from foam-framework/foam2#368 and #43 (which introduces support for `serializer;`

**Comment by mdittmer** [comment]
@arobins and we're green :raised_hands:

**Review by arobins — APPROVED** [review]
lgtm

**Pull request merged**